src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,33 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate firstDate = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate lastDate = LocalDate.parse(dateTo, FORMATTER);
+        int[] money = new int[names.length];
+        StringBuilder builder = new StringBuilder();
+        builder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        for (String newData : data) {
+            String[] salaryPerDay = newData.split(" ");
+            LocalDate dataDate = LocalDate.parse(salaryPerDay[0], FORMATTER);
+            if (!(dataDate.isBefore(firstDate)) && !(dataDate.isAfter(lastDate))) {
```

Comment:
```suggestion
            if (dataDate.isBefore(lastDate)  && dataDate.isAfter(firstDate)) {
```
it looks simpler this way